app/views/courses/HeroSelectModal.coffee

```diff
@@ -0,0 +1,42 @@
+ModalView = require 'views/core/ModalView'
+template = require 'templates/courses/hero-select-modal'
+Classroom = require 'models/Classroom'
+CocoCollection = require 'collections/CocoCollection'
+State = require 'models/State'
+ThangType = require 'models/ThangType'
+User = require 'models/User'
+
+module.exports = class HeroSelectModal extends ModalView
+  id: 'hero-select-modal'
+  template: template
+
+  events:
+    'click .select-hero-btn': 'onClickSelectHeroButton'
+    'click .hero-option': 'onClickHeroOption'
+
+  initialize: ({ currentHeroID }) ->
+    @state = new State({
+      currentHeroID
+      selectedHeroID: currentHeroID
+    })
+
+    @heroes = new CocoCollection([], {model: ThangType})
+    @heroes.url = '/db/thang.type?view=heroes'
+    @heroes.setProjection ['original', 'name', 'heroClass']
+    @heroes.comparator = 'gems' # TODO: Random? Alphabetical? Something else?
+    @supermodel.loadCollection(@heroes, 'heroes')
+
+    @listenTo @state, 'all', @render
+    @listenTo @heroes, 'all', @render
+
+  onClickHeroOption: (e) ->
+    heroID = $(e.currentTarget).data('hero-id')
+    @state.set selectedHeroID: heroID
+    hero = @heroes.get(heroID)
+    me.set(heroConfig: {}) unless me.get('heroConfig')
+    me.get('heroConfig').thangType = hero.get('original')
+    me.save().then =>
+      @trigger 'hero-select:success', hero
```

Comment:
Should follow forms best practices doc:

> * On form submitted, disable submit button and show “saving”, “sending”, etc. Hide any alerts and validation states.
> * On user form errors, if the form is small and the error is field-specific, use help text (above any existing help text) and Bootstrap form validation states. Otherwise, show an error alert above the submit button. Auto scroll to the topmost error.

So, disable and show 'saving' on the button, and handle network errors. And shouldn't the system save when you click the select hero button, not when you click a hero icon? What happens if a user clicks a hero and then clicks the 'x' at the top? My expectation would be that the change was cancelled.